R/plotDispersion.R

```diff
@@ -85,59 +166,65 @@ plotDispersion <- function(DGEdata,
-                                                 afterRender             = afterRender)
+        tryCatch({
+            MyDispPlot <- canvasXpress::canvasXpress(data                    = plotdata,
+                                                    graphType               = "Scatter2D",
+                                                    colors                  = symbolColor,
+                                                    dataPointSize           = symbolSize,
+                                                    shapes                  = symbolShape,
+                                                    title                   = title,
+                                                    yAxisTitle              = ylab,
+                                                    showLoessFit            = showLoessFit,
+                                                    fitLineColor            = linefitColor,
+                                                    fitLineStyle            = lineType,
+                                                    afterRender             = afterRender)
+        },
+        error = function(e) {
+            warning('Unable to render the canvasXpress plot:', e)
+        })
     } else {
-        MyDispPlot <- ggplot(plotdata, aes(x = AveLogCPM, y = Dispersion)) +
-            geom_point(size = symbolSize,
-                       shape = symbolShape,
-                       fill = symbolFill,
-                       color = symbolColor,
-                       alpha = symbolAlpha)
+        tryCatch({
+            MyDispPlot <- ggplot(plotdata, aes(x = AveLogCPM, y = Dispersion)) +
+                geom_point(size  = symbolSize,
+                           shape = symbolShape,
+                           fill  = symbolFill,
+                           color = symbolColor,
+                           alpha = symbolAlpha)
 
-        if (!is.null(lineFit)) {
-            MyDispPlot <- MyDispPlot +
-                geom_smooth(method = lineFit,
-                            size = linefitSize,
-                            color = linefitColor,
-                            linetype = lineType,
-                            alpha = lineAlpha)
-        }
+            if (!is.null(lineFit)) {
+                MyDispPlot <- MyDispPlot +
+                    geom_smooth(formula  = y ~ x,
+                                method   = lineFit,
+                                color    = linefitColor,
+                                linetype = lineType)
+            }
 
-        MyDispPlot <- MyDispPlot +
-            ylab(ylab) +
-            ggtitle(title) +
-            expand_limits(x = 0, y = 0)
+            MyDispPlot <- MyDispPlot +
+                ylab(ylab) +
+                ggtitle(title) +
+                expand_limits(x = 0, y = 0)
+        },
+        error = function(e) {
+            warning('Unable to render ggplot:', e)
+        })
     }
 
     return(MyDispPlot)
```

Comment:
In your **tryCatch** statements ad a finally statement to return **MyDispPlot** to avoid using the **return** method at the end (Connie does not liked it here)
Also you can (and should) return your output just by mentioned **MyDispPlot** without explicit call for **return** method

Reply:
I want to check this with you. The try catch does not help here. Exception only gets triggered while evaluating myDispPlot. I am removing for now. Would like to discuss this with you.

Reply:
Sure, any time